Buildings/HeatTransfer/Conduction/SingleLayer.mo

```diff
@@ -32,30 +45,66 @@ model SingleLayer "Model for single layer heat conductance"
+     annotation (Evaluate=true);
 protected
-  final parameter Integer nSta(min=1) = material.nSta
+  final parameter Integer nSta=
+    max(nSta2,
+        if placeCapacityAtSurf_a or placeCapacityAtSurf_b then 2 else 1)
     "Number of state variables";
-  final parameter Modelica.SIunits.ThermalResistance RnSta = R/nSta
-    "Thermal resistance between nodes";
-  final parameter Modelica.SIunits.ThermalResistance RnSta2 = RnSta/2
-    "Thermal resistance between nodes and surface boundary";
+  final parameter Integer nR=nSta+1 "Number of thermal resistances";
+  parameter Modelica.SIunits.ThermalResistance RNod[nR]=
+    if (placeCapacityAtSurf_a and placeCapacityAtSurf_b) then
+      if (nSta==2) then
+        {(if i==1 or i==nR then 0 else R/(nSta-1)) for i in 1:nR}
+      else
+        {(if i==1 or i==nR then 0 elseif i==2 or i==nR-1 then R/(2*(nSta-2)) else R/(nSta-2)) for i in 1:nR}
+      elseif (placeCapacityAtSurf_a and (not placeCapacityAtSurf_b)) then
+        {(if i==1 then 0 elseif i==2 or i==nR then R/(2*(nSta-1)) else R/(nSta-1)) for i in 1:nR}
+    elseif (placeCapacityAtSurf_b and (not placeCapacityAtSurf_a)) then
+       {(if i==nR then 0 elseif i==1 or i==nR-1 then R/(2*(nSta-1)) else R/(nSta-1)) for i in 1:nR}
+    else
+      {R/(if i==1 or i==nR then (2*nSta) else nSta) for i in 1:nR}
+    "Thermal resistance";
 
-  parameter Modelica.SIunits.Mass m = A*material.x*material.d/material.nSta
+  parameter Modelica.SIunits.Mass m[nSta]=
+   (A*material.x*material.d) *
+   (if (placeCapacityAtSurf_a and placeCapacityAtSurf_b) then
+     if (nSta==2) then
+       {1/(2*(nSta-1)) for i in 1:nSta}
+     elseif (nSta==3) then
+       {1/(if i==1 or i==nSta then (2*(nSta-1)) else (nSta-1)) for i in 1:nSta}
+     else
+       {1/(if i==1 or i==nSta or i==2 or i==nSta-1 then (2*(nSta-2)) else (nSta-2)) for i in 1:nSta}
+     elseif (placeCapacityAtSurf_a and (not placeCapacityAtSurf_b)) then
+       {1/(if i==1 or i==2 then (2*(nSta-1)) else (nSta-1)) for i in 1:nSta}
+     elseif (placeCapacityAtSurf_b and (not placeCapacityAtSurf_a)) then
+       {1/(if i==nSta or i==nSta-1 then (2*(nSta-1)) else (nSta-1)) for i in 1:nSta}
+     else
+       {1/(nSta) for i in 1:nSta})
     "Mass associated with the temperature state";
-  final parameter Modelica.SIunits.HeatCapacity C = m*material.c
+
+  final parameter Modelica.SIunits.HeatCapacity C[nSta] = m*material.c
     "Heat capacity associated with the temperature state";
-  final parameter Real CInv = if material.steadyState then 0 else 1/C
+  final parameter Real CInv[nSta]=
+    if material.steadyState then zeros(nSta) else {1/C[i] for i in 1:nSta}
     "Inverse of heat capacity associated with the temperature state";
 
-  parameter Modelica.SIunits.SpecificInternalEnergy ud[Buildings.HeatTransfer.Conduction.nSupPCM](each fixed=false)
+  parameter Modelica.SIunits.SpecificInternalEnergy ud[Buildings.HeatTransfer.Conduction.nSupPCM](
```

Comment:
@mwetter `Buildings.HeatTransfer.Conduction.nSupPCM` which is the number of support points for derivatives is a constant so I am not sure how this relates to `nSta`?